src/libraries/Common/src/System/Net/CookieParser.cs

```diff
@@ -44,13 +45,13 @@ internal class CookieTokenizer
         private bool _eofCookie;
         private int _index;
         private readonly int _length;
-        private string _name;
+        private string? _name;
         private bool _quoted;
         private int _start;
         private CookieToken _token;
         private int _tokenLength;
         private readonly string _tokenStream;
-        private string _value;
+        private string _value = string.Empty;
```

Comment:
Oh, It is assigned only once from `internal string Extract()` method and design is that the value is never null. Maybe we accept the change? Otherwise there will be many changes in the code without any benefits.

Reply:
To me this is a similar case to IEnumerator.MoveNext: the contract is that you should access Current until after you've called MoveNext, and thus we don't consider the nullability contracts to be valid if you access it outside of that pattern.  Here, instead of MoveNext, it's Next, and instead of Current, it's Value.  So it seems reasonable to say that `_value` is non-nullable, and initialize it here to `null!` because it'll only actually be accessed after Next has been called, and Next will initialize it to non-null.

Reply:
Fixed.